tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/tests/MockListener.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.nativeimpl.jvm.tests;
+
+import org.ballerinalang.jvm.api.BExecutor;
+import org.ballerinalang.jvm.api.connector.CallableUnitCallback;
+import org.ballerinalang.jvm.api.values.BError;
+import org.ballerinalang.jvm.api.values.BObject;
+import org.ballerinalang.jvm.api.values.BString;
+import org.ballerinalang.jvm.scheduling.Scheduler;
+
+import java.util.HashMap;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * A mock listener for testing services. It can be used to invoke a resource in the service.
+ */
+public class MockListener {
+
+    private static BObject service;
+    private static BError err;
+
+    public static Object attach(BObject servObj) {
+        service = servObj;
+        return null;
+    }
+
+    public static Object invokeResource(BString name) throws InterruptedException {
+        if (service != null) {
+            CountDownLatch latch = new CountDownLatch(1);
+            BExecutor.submit(Scheduler.getStrand().scheduler, service, name.getValue(), null, null,
+                             new CallableUnitCallback() {
+                                 @Override
+                                 public void notifySuccess() {
+                                     latch.countDown();
+                                 }
+
+                                 @Override
+                                 public void notifyFailure(BError error) {
+                                     err = error;
+                                     latch.countDown();
+                                 }
+                             }, new HashMap<>(), null);
+            latch.await();
```

Comment:
Can you use the BRuntime api's to do this invocation? BExecutor will be removed.

Reply:
The BRuntime apis allow executing functions. Can they also be used for executing resources? I found the http code uses the above function

Reply:
Yes can be done. Thats what @manuranga is changing and http/stdlib will not be using this BExecutor api's anymore and they will be migrated to use BRuntime api's. 

Check the `invokeMethodAsync` method in BRuntime. It should do the above.

Reply:
And this class BExecutor is no longer there as it has been just removed :)

Reply:
I have fixed this